### PR TITLE
Add table aws_mgn_application closes #1454

### LINKED ISSUE #1454: Add table aws_mgn_application
**References**
https://github.com/aws/aws-sdk-go-v2/blob/main/service/mgn/api_op_ListApplications.go#L16

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select
  name,
  application_id,
  creation_date_time,
  is_archived,
  wave_id
from
  aws_mgn_application
where
  creation_date_time >= now() - interval '30' day;
+-------+-----------------------+---------------------------+-------------+------------------------+
| name  | application_id        | creation_date_time        | is_archived | wave_id                |
+-------+-----------------------+---------------------------+-------------+------------------------+
| test2 | app-3c0e4c72641c56b44 | 2022-12-16T14:01:46+05:30 | false       | wave-39e09a6f5a86f6ad3 |
+-------+-----------------------+---------------------------+-------------+------------------------+

> select
  name,
  application_id,
  application_aggregated_status ->> 'HealthStatus' as health_status,
  application_aggregated_status ->> 'ProgressStatus' as progress_status,
  application_aggregated_status ->> 'TotalSourceServers' as total_source_servers
from
  aws_mgn_application;
+-------+-----------------------+---------------+-----------------+----------------------+
| name  | application_id        | health_status | progress_status | total_source_servers |
+-------+-----------------------+---------------+-----------------+----------------------+
| test2 | app-3c0e4c72641c56b44 | HEALTHY       | NOT_STARTED     | 0                    |
+-------+-----------------------+---------------+-----------------+----------------------+

```
</details>

### CODE REVIEW

File: aws/table_aws_mgn_application.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+//// TABLE DEFINATION
+
+func tableAwsMGNApplication(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_mgn_application",
+		Description: "AWS MGN Application",
+		List: &plugin.ListConfig{
+			Hydrate: ListAwsMGNApplication,
+			IgnoreConfig: &plugin.IgnoreConfig{
+				// UninitializedAccountException - This error comes up when default template are not set for a particular region.
```
Comment (misraved) [inline aws/table_aws_mgn_application.go:26]: Can we provide a doc link for this error?

Comment (ParthaI) [reply aws/table_aws_mgn_application.go:26]: I didn't find any doc for this error, Figured this out during the testing.

File: aws/table_aws_mgn_application.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+)
+
+//// TABLE DEFINATION
+
+func tableAwsMGNApplication(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_mgn_application",
+		Description: "AWS MGN Application",
+		List: &plugin.ListConfig{
+			Hydrate: ListAwsMGNApplication,
```
Comment (misraved) [inline aws/table_aws_mgn_application.go:24]:
```suggestion
			Hydrate: ListAwsMGNApplications,
```

File: aws/table_aws_mgn_application.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+			IgnoreConfig: &plugin.IgnoreConfig{
+				// UninitializedAccountException - This error comes up when default template are not set for a particular region.
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"UninitializedAccountException"}),
+			},
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "application_id", Require: plugin.Optional},
+				{Name: "wave_id", Require: plugin.Optional},
+				{Name: "is_archived", Require: plugin.Optional, Operators: []string{"=", "<>"}},
```
Comment (misraved) [inline aws/table_aws_mgn_application.go:32]: Can a boolean column be used as an optional qual? What would happen if the user uses the following query - 
`select * from aws_mgn_application where is_archived`?

File: aws/table_aws_mgn_application.go [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+		plugin.Logger(ctx).Error("aws_mgn_application.ListAwsMGNApplication", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupport region, return no data
+		return nil, nil
+	}
+
+	// Reduce the basic request limit down if the user has only requested a small number of rows
+	maxLimit := int32(100)
```
Comment (misraved) [inline aws/table_aws_mgn_application.go:120]:
```suggestion
	maxLimit := int32(1000)
```
As per the documentation, I think the max limit should be `1000` - https://docs.aws.amazon.com/mgn/latest/APIReference/API_ListApplications.html

File: aws/table_aws_mgn_application.go [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+
+// // LIST FUNCTION
+func ListAwsMGNApplication(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	svc, err := MGNClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_mgn_application.ListAwsMGNApplication", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupport region, return no data
```
Comment (misraved) [inline aws/table_aws_mgn_application.go:115]:
```suggestion
		// Unsupported region, return no data
```

File: aws/table_aws_mgn_application.go [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Arn").Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+// // LIST FUNCTION
+func ListAwsMGNApplication(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (misraved) [inline aws/table_aws_mgn_application.go:108]:
```suggestion
func ListAwsMGNApplications(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

Comment (misraved) [reply aws/table_aws_mgn_application.go:108]: Please update all instances of `ListAwsMGNApplication` to `ListAwsMGNApplications`

File: docs/tables/aws_mgn_application.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
+# Table: aws_mgn_application
+
+AWS Application Migration Service (MGN) is a highly automated lift-and-shift (rehost) solution that simplifies, expedites, and reduces the cost of migrating applications to AWS. It enables companies to lift-and-shift a large number of physical, virtual, or cloud servers without compatibility issues, performance disruption, or long cutover windows.
+
```
Comment (misraved) [inline docs/tables/aws_mgn_application.md:4]:
```suggestion
AWS Application Migration Service (MGN) is a highly automated lift-and-shift (rehost) solution that simplifies, expedites, and reduces the cost of migrating applications to AWS. It enables companies to lift and shift many physical, virtual, or cloud servers without compatibility issues, performance disruption, or long cutover windows.

```

File: docs/tables/aws_mgn_application.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+  name,
+  application_id,
+  application_aggregated_status ->> 'HealthStatus' as health_status,
+  application_aggregated_status ->> 'ProgressStatus' as progress_status,
+  application_aggregated_status ->> 'TotalSourceServers' as total_source_servers
+from
+  aws_mgn_application;
+```
+
+### List application that are created in last 30 days
```
Comment (misraved) [inline docs/tables/aws_mgn_application.md:50]:
```suggestion
### List applications that have been created in the last 30 days
```

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments. Thanks!!

**Review by misraved — APPROVED** [review]

**Pull request merged**